### PR TITLE
Floating Menu: Allow fixing / disabling the element menu

### LINKED ISSUE #11570: Floating Menu: Consider ability to disable or change the default positioning
## Feature Description

Users that are creating many web stories daily (power users) are unhappy with the floating toolbar introduced in v1.19 of the plugin, and have requested the ability to :
1. disable the presentation of the toolbar for individual users
2. move the default location of the floating toolbar to another, consistent location on the canvas.

### PR DESCRIPTION
**Pull request by miina**
## Summary
Adds a new Settings menu to the Element toolbar/floating menu that allows moving the menu to the top / to the element / allows persistently closing the toolbar.

## Relevant Technical Choices
- Increases the header height a bit to fit the toolbar

## To-do

- [x] Confirm UX
- [ ] Add icon from UX / update copy
- [x] Tests

## User-facing changes
Screenshot 2022-08-17 at 17 38 58
Screenshot 2022-08-17 at 17 38 51

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Add an element
2. See the new Settings menu in the floating menu.
3. Verify that it's possible to place the element above the page using the "Fix to top" option and back to the element with the "Fix to element" option.
4. Verify that closing the menu from "Always hide" will persistently hide the menu (deselecting and selecting the element again will not reopen the menu)
5. Verify that when closing the menu from "Always hide", a new button appears to the bottom right area and that button is highlighted after closing the menu.
6. Verify that the button opens the menu again

### Does this PR have a security-related impact?
No

### Does this PR change what data or activity we track or use?
```
trackEvent('floating_menu', {
      name: 'reposition_menu',
      position,
    });

```
```
trackEvent('floating_menu', {
      name: 'persistent_menu_hide',
    });
```

### Does this PR have a legal-related impact?
No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11570

### CODE REVIEW

**Comment by miina** [comment]
@barklund This is still waiting for the UX confirmation but the general logic should be in place, would be great if you could review already.

File: packages/story-editor/src/components/floatingMenu/constants.js [hunk 1]
```diff
@@ -28,6 +28,11 @@ export const SELECTED_ELEMENT_TYPES = {
   NONE: 'none',
 };
 
+export const TOOLBAR_POSITIONS = {
+  element: 'fixed_to_element',
+  top: 'fixed_to_top',
+};
```
Comment (merapi) [inline packages/story-editor/src/components/floatingMenu/constants.js]: **🥜 nitpick:** I think we use uppercase for nested const variables more often than not, albeit we are not consistent 😉 

image

**Comment by merapi** [comment]
Looks good, but UX-wise: without the video, I wouldn't notice the "Toggle element toolbar" button - let's talk about it on the design sync.

**Comment by miina** [comment]
This is now ready -- only waiting for UX to switch the Toolbar icon + update the copy.

**Review by barklund — APPROVED** [review]
This looks good - I 🕺  this!

**Review by merapi — APPROVED** [review]
👍🏻 

I centered the icon.

**Pull request merged**